src/text-filter/__tests__/use-debounce-search-result-callback.test.tsx

```diff
@@ -0,0 +1,125 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+import { render } from '@testing-library/react';
+
+import useDebounceSearchResultCallback, {
+  DebouncedLiveAnnouncementProps,
+} from '../../../lib/components/text-filter/use-debounce-search-result-callback';
+
+const TestComponent = (props: DebouncedLiveAnnouncementProps) => {
+  useDebounceSearchResultCallback(props);
+  return null;
+};
+
+const renderTestComponent = (props: DebouncedLiveAnnouncementProps) => {
+  const { rerender } = render(<TestComponent {...props} />);
+  jest.runAllTimers();
+
+  return {
+    rerender,
+  };
+};
+
+describe('useDebounceSearchResultCallback', () => {
+  let announceCallback: () => void;
+  beforeEach(() => {
+    announceCallback = jest.fn();
+    jest.useFakeTimers();
+  });
+
+  afterEach(() => {
+    jest.useRealTimers();
+  });
+
+  describe('sync filtering (loading = false)', () => {
+    test('should not call announceCallback when searchQuery and countText are undefined', () => {
+      renderTestComponent({ announceCallback, loading: false, searchQuery: undefined, countText: undefined });
+      expect(announceCallback).not.toHaveBeenCalled();
+    });
+
+    test('should not call announceCallback when searchQuery is present and countText is undefined', () => {
+      renderTestComponent({ announceCallback, loading: false, searchQuery: '', countText: undefined });
+      expect(announceCallback).not.toHaveBeenCalled();
+    });
+
+    test('should call announceCallback when searchQuery and countText are present', () => {
+      renderTestComponent({ announceCallback, loading: false, searchQuery: 'A', countText: '123 matches' });
+      expect(announceCallback).toHaveBeenCalledTimes(1);
+    });
+
+    test('should call announceCallback twice when searchQuery changes and countText stays the same', () => {
```

Comment:
Maybe we can drop **twice** from the test title Because the test purpose is not to make sure it calls it twice, but more like to make sure it is announced when the search term is changing and the count text is unchanged.

Reply:
Addressed - Right, I've updated the test name + added an additional assertion after the first render to ensure the announcement is called on the initial render + rerender as expected.